include/deal.II/lac/arpack_solver.h

```diff
@@ -564,14 +633,18 @@ ArpackSolver::solve(const MatrixType1 & /*system_matrix*/,
          ArpackExcSmallNumberofArnoldiVectors(
            additional_data.number_of_arnoldi_vectors, nev));
 
-  // ARPACK mode for dsaupd/dnaupd, here only mode 3, i.e. shift-invert mode
-  int mode = 3;
+  // ARPACK mode for dsaupd/dnaupd, here only mode 1 (exact-shifting) and 3,
+  // i.e. shift-invert mode
+  int mode = additional_data.mode;
 
   // reverse communication parameter
   int ido = 0;
 
   // 'G' generalized eigenvalue problem 'I' standard eigenvalue problem
-  char bmat[2] = "G";
+  char bmat[2] = {eigenvalue_problem_type};
```

Comment:
can you also explicitly initialize the second element

Reply:
As far as I know the second element is never used. I assume that this was done for compatibility with Fortran.

Reply:
Right, but right now it's left on an undefined value. It was previously `"G"` which is `{'G', '\0'}`.